src/sentry/static/sentry/app/components/dataPrivacyRules/eventIdField.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import isEqual from 'lodash/isEqual';
+
+import TextField from 'app/components/forms/textField';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+
+import {FormField} from './formField';
+import EventIdFieldStatusIcon from './eventIdFieldStatusIcon';
+import {EventIdStatus, EventId} from './types';
+
+type Props = {
+  onUpdateEventId: (eventId: string) => void;
+  eventId?: EventId;
+  disabled?: boolean;
+};
+
+type State = {
+  value: string;
+  status?: EventIdStatus;
+};
+
+class EventIdField extends React.Component<Props, State> {
+  state = {
+    value: this.props.eventId?.value || '',
+    status: this.props.eventId?.status,
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (!isEqual(prevProps.eventId, this.props.eventId)) {
+      this.loadState();
+    }
+  }
+
+  loadState = () => {
+    this.setState({
+      ...this.props.eventId,
+    });
+  };
+
+  handleChange = (value: string) => {
+    const eventId = value.replace(/-/g, '').trim();
+
+    if (eventId !== this.state.value) {
+      this.setState({
+        value: eventId,
+        status: undefined,
+      });
+    }
+  };
+
+  isEventIdValid = (): boolean => {
+    const {value} = this.state;
+
+    if (value && value.length !== 32) {
+      this.setState({status: EventIdStatus.INVALID});
+      return false;
+    }
+
+    return true;
+  };
+
+  handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    if (this.isEventIdValid()) {
+      this.props.onUpdateEventId(this.state.value);
+    }
+  };
+
+  handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const {keyCode} = event;
+
+    if (keyCode === 13 && this.isEventIdValid()) {
+      this.props.onUpdateEventId(this.state.value);
+    }
+  };
+
+  handleClickIconClose = () => {
+    this.setState({
+      value: '',
+      status: undefined,
+    });
+  };
+
+  getErrorMessage = (): string | undefined => {
+    const {status} = this.state;
+
+    switch (status) {
+      case EventIdStatus.INVALID:
+        return t('This event ID is invalid.');
+      case EventIdStatus.ERROR:
+        return t(
+          'An error occurred while fetching the suggestions based on this Event ID.'
+        );
+      case EventIdStatus.NOT_FOUND:
+        return t('The chosen event ID was not found in projects you have access to.');
```

Comment:
```suggestion
        return t('The chosen Event ID was not found in projects you have access to.');
```

Event ID be capitalized consistently if we're going to title case it.